src/khoj/app/settings.py

```diff
@@ -112,20 +114,95 @@
     },
 ]
 
-ASGI_APPLICATION = "app.asgi.application"
 
 CLOSE_CONNECTIONS_AFTER_REQUEST = True
 
 # Database
 # https://docs.djangoproject.com/en/4.2/ref/settings/#databases
 DATA_UPLOAD_MAX_NUMBER_FIELDS = 20000
+
+# Default PostgreSQL configuration
+DB_NAME = os.getenv("POSTGRES_DB", "khoj")
+DB_HOST = os.getenv("POSTGRES_HOST", "localhost")
+DB_PORT = os.getenv("POSTGRES_PORT", "5432")
+
+# Use pgserver if env var explicitly set to true
+USE_PGSERVER = is_env_var_true("POSTGRES_USE_PGSERVER")
+
+if USE_PGSERVER:
+    # Set up logging for pgserver
+    logger = logging.getLogger("pgserver_django")
+    logger.setLevel(logging.INFO)
+    if not logger.handlers:
+        handler = logging.StreamHandler()
+        handler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))
+        logger.addHandler(handler)
+
+    WSGI_APPLICATION = "app.wsgi.application"
```

Comment:
Why do you need to do WSGI with `pg_server`? It doesn't work with the async server? This might have interesting consequences on the open connections issue, namely that it might be a mitigation.

Reply:
Previously thought the pg server is not working as an ASGI app. But on checking again, it is not required to be a WSGI app. So reverted that portion